packages/components/pages/f-mfa/src/components/_tests/Mfa.test.js

```diff
@@ -80,36 +106,195 @@ describe('Mfa', () => {
+                    ...routeMock,
+                    query: {
+                        ...routeMock.query,
+                        [key]: value
+                    }
+                }
+            };
+
+            // Act
+            wrapper = await mountSut({ mocks: sutOverrideMocks });
+
+            // Assert
+            const errorCard = wrapper.find('[data-test-id="mfa-error-page"]');
+            expect(errorCard.exists()).toBe(expected);
+            expect(warnLogSpy).toHaveBeenCalled();
+        });
+
+        it.each([
+            ['returnUrl'],
+            ['email'],
+            ['code']
+        ])('should show the error page and log warn if the %s values are missing', async key => {
+            // Arrange
+            const expected = true;
+            const queryMock = {
+                ...routeMock.query
+            };
+            delete queryMock[key];
+            const sutOverrideMocks = {
+                ...sutMocks,
+                $route: {
+                    ...routeMock,
+                    query: queryMock
+                }
+            };
+
+            // Act
+            wrapper = await mountSut({ mocks: sutOverrideMocks });
+
+            // Assert
+            const errorCard = wrapper.find('[data-test-id="mfa-error-page"]');
+            expect(errorCard.exists()).toBe(expected);
+            expect(warnLogSpy).toHaveBeenCalled();
+        });
+    });
+
+    describe('When calling the onFormSubmit() method', () => {
+        it.each([
+            ['show', true],
+            ['NOT show', false]
+        ])('should %s error alert when hasSubmitError is %s', async (_, hasSubmitError) => {
+            // Arrange
+            wrapper = await mountSut();
+
+            // Act
+            await wrapper.setData({ hasSubmitError });
+
+            // Assert
+            const alert = wrapper.find('[data-test-id="mfa-submit-error-alert"]');
+            expect(alert.exists()).toBe(hasSubmitError);
+        });
+
+        it.each([
+            ['show', true],
+            ['NOT show', false]
+        ])('should %s validation error when showValidationError is %s', async (_, showValidationError) => {
+            // Arrange
+            wrapper = await mountSut({ shallow: false }); // Deep mount to make sure slot is rendered
+
+            // Act
+            await wrapper.setData({ showValidationError });
+
+            // Assert
+            const validationError = wrapper.find('[data-test-id="mfa-form-validation-error"]');
+            expect(validationError.exists()).toBe(showValidationError);
+        });
+
+        describe('And the form data is valid', () => {
+            it('should call the postValidateMfaToken() method with the correct params', async () => {
+                // Arrange
+                mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
+                const expectedParams = { mfa_token: 'ABC123', otp: 'test-otp' }; // eslint-disable-line camelcase
```

Comment:
Is the `mfa_token` needed for this test? Should it use the routeMock value by default?